### ISSUE TITLE
Feature Enhancement: Expand Regression Line Capabilities in graph_utils

### ISSUE THREAD

**Issue by 09hanvaik** [Data-Simply/pyretailscience#318]
# 🚀 Feature Enhancement: Expand Regression Line Capabilities in graph_utils

  ## 📊 Overview
  Enhance the existing `add_regression_line()` function in 
  `pyretailscience/plots/styles/graph_utils.py` to support multiple regression types beyond the
  current linear regression, providing comprehensive trend analysis capabilities for retail data
  visualization.

  ## 🎯 Current State
  - ✅ Linear regression implemented with equation display and R² values
  - ✅ Works with line and scatter plots
  - ✅ Handles numeric and datetime x-axis values
  - ✅ Configurable styling and text positioning

  ## 📋 Implementation Priorities

  ### 🔥 Priority 1: Core Improvements
  **Goal**: Stabilize and enhance existing linear regression functionality

  #### Acceptance Criteria:
  - [ ] **Default to current linear regression** when no `regression_type` parameter specified
  - [ ] **Backward compatibility**: All existing calls to `add_regression_line()` continue to work        
  unchanged
  - [ ] **Enhanced parameter validation** with clear error messages
  - [ ] **Improved equation formatting** for better readability
  - [ ] **Unit tests** covering current linear regression functionality
  - [ ] **Documentation updates** with usage examples

  #### Technical Requirements:
  ```python
  def add_regression_line(
      ax: Axes,
      regression_type: str = "linear",  # NEW: Default to current behavior
      color: str = "red",
      linestyle: str = "--",
      text_position: float = 0.6,
      show_equation: bool = True,
      show_r2: bool = True,
      **kwargs: dict[str, any],
  ) -> Axes:

  🚀 Priority 2: Core Regression Types

  Goal: Add essential regression types for retail analytics

  Acceptance Criteria:

  - Linear Regression (existing, enhanced)
    - Equation: y = mx + b
    - R² calculation and display
  - Power Law Regression (Log-Log, Constant Elasticity)
    - Equation: y = ax^b → log(y) = log(a) + b*log(x)
    - Elasticity interpretation for price-demand analysis
    - Handle zero/negative values gracefully
  - Logarithmic Regression
    - Equation: y = a*ln(x) + b
    - Useful for diminishing returns analysis
  - Exponential Decay Regression
    - Equation: y = ae^(bx) → ln(y) = ln(a) + bx
    - Customer retention, inventory decay patterns

  Technical Requirements:

  # Usage examples:
  add_regression_line(ax, regression_type="linear")       # Default
  add_regression_line(ax, regression_type="power")        # y = ax^b
  add_regression_line(ax, regression_type="logarithmic")  # y = a*ln(x) + b
  add_regression_line(ax, regression_type="exponential")  # y = ae^(bx)

  Test Coverage:

  - Unit tests for each regression type
  - Edge cases: Zero/negative values, insufficient data points
  - Accuracy tests: Known datasets with expected coefficients
  - Visual regression tests: Equation display formatting

  ⭐ Priority 3: Advanced Features

  Goal: Power-user capabilities for complex analysis

  Acceptance Criteria:

  - Polynomial Regression (orders 2-6)
    - Equation: y = a₀ + a₁x + a₂x² + ... + aₙxⁿ
    - Configurable polynomial degree
    - Cross-validation for optimal degree selection
  - Segmented Regression (Piecewise Linear)
    - Automatic breakpoint detection
    - Multiple linear segments with different slopes
  - Robust Regression
    - RANSAC for outlier resistance
    - Huber regression for heavy-tailed errors

  Advanced Configuration:

  add_regression_line(ax, regression_type="polynomial", degree=3)
  add_regression_line(ax, regression_type="segmented", max_segments=2)
  add_regression_line(ax, regression_type="robust", method="ransac")

  Analytics Features:

  - Confidence intervals display option
  - Prediction intervals for forecasting
  - Residual analysis integration
  - Model comparison metrics (AIC, BIC)
  - Cross-validation R² scores

  🛠️ Technical Specifications

  Recommended Dependencies:

  # Required
  import numpy as np
  from scipy import stats
  from scipy.optimize import curve_fit
  from sklearn.preprocessing import PolynomialFeatures
  from sklearn.linear_model import LinearRegression, RANSACRegressor, HuberRegressor

  Error Handling Recommendations:

  - Graceful handling of non-convergent fits
  - Clear error messages for inappropriate data types
  - Fallback to linear regression for failed complex fits

  Performance Considerations:

  - Efficient algorithms for large datasets (>10k points)
  - Caching of computationally expensive fits
  - Optional progress indicators for slow operations

  📚 Business Value

  Retail Analytics Use Cases:

  1. Price Elasticity Analysis → Power law regression
  2. Customer Lifetime Value → Exponential decay
  3. Seasonal Demand Patterns → Polynomial fitting
  4. Marketing Response Curves → Logarithmic regression
  5. Inventory Turnover → Linear/exponential trends

  Expected Impact:

  - Enhanced Decision Making: More accurate trend analysis
  - Improved Forecasting: Better model selection capabilities
  - Deeper Insights: Advanced regression types reveal hidden patterns
  - Time Savings: Built-in analytics reduce external tool dependencies

  🧪 Testing Strategy Recommendations

  Suggested Test Categories:

  1. Unit Tests: Each regression type with known datasets
  2. Integration Tests: Full plotting pipeline with real retail data
  3. Performance Tests: Large dataset handling (>10k points)
  4. Visual Tests: Equation formatting and positioning
  5. Edge Case Tests: Corner cases and error conditions

  Recommended Test Data:

  - Synthetic datasets with known parameters
  - Real retail transaction data (anonymized)
  - Edge cases: zeros, negatives, missing values, outliers

  📖 Documentation Recommendations

  Suggested Updates:

  - Function docstrings with regression type explanations
  - Usage examples for each regression type
  - Mathematical background for business users
  - Performance benchmarks and recommendations
  - Migration guide for existing code

  Example Usage:

  # Business Example: Price Elasticity Analysis
  import pyretailscience.plots.styles.graph_utils as gu

  # Power law regression for price-demand relationship
  ax = price_demand_df.plot(x='price', y='demand', kind='scatter')
  gu.add_regression_line(ax, regression_type="power",
                        show_equation=True, show_r2=True)
  # Displays: y = 1000x^(-1.2), R² = 0.94

  🏷️ Suggested Labels

  - enhancement
  - plotting
  - analytics
  - regression
  - graph-utils
  - priority-medium

  💡 Additional Recommendations

  - Breaking Changes: None (fully backward compatible)
  - Dependencies: Self-contained enhancement
  - Implementation Approach: Incremental by priority level
